tests/kafkatest/tests/streams/streams_multiple_rolling_upgrade_test.py

```diff
@@ -0,0 +1,110 @@
+from kafkatest.tests.streams.base_streams_test import BaseStreamsTest
+from kafkatest.version import LATEST_0_10_2, LATEST_0_11_0, LATEST_1_0,  DEV_BRANCH, KafkaVersion
+
+
+class StreamsMultipleRollingUpgradeTest(BaseStreamsTest):
+    """
+     This test will verify a rolling upgrade of multiple streams
+     applications against all versions of streams against a single
+     broker version.
+
+     As new releases come out, just update the streams_upgrade_versions array to have the latest version
+     included in the list.
+
+     A prerequisite for this test to succeed
+     is the inclusion of all parametrized versions of kafka in kafka/vagrant/base.sh
+     (search for get_kafka()).
+     As new versions are released the kafka/tests/kafkatest/version.py file
+     needs to be updated as well.
+
+     You can find what's been uploaded to S3 with the following command
+
+     aws s3api list-objects --bucket kafka-packages --query 'Contents[].{Key:Key}
+    """
+    # adding new version to this list will cover broker and streams version
+    streams_upgrade_versions = [str(LATEST_0_10_2), str(LATEST_0_11_0), str(LATEST_1_0), str(DEV_BRANCH)]
+
+    def __init__(self, test_context):
+        super(StreamsMultipleRollingUpgradeTest, self).__init__(test_context,
+                                                                topics={
+                                                                    'echo': {'partitions': 5, 'replication-factor': 1},
+                                                                    'data': {'partitions': 5, 'replication-factor': 1},
+                                                                    'min': {'partitions': 5, 'replication-factor': 1},
+                                                                    'max': {'partitions': 5, 'replication-factor': 1},
+                                                                    'sum': {'partitions': 5, 'replication-factor': 1},
+                                                                    'dif': {'partitions': 5, 'replication-factor': 1},
+                                                                    'cnt': {'partitions': 5, 'replication-factor': 1},
+                                                                    'avg': {'partitions': 5, 'replication-factor': 1},
+                                                                    'wcnt': {'partitions': 5, 'replication-factor': 1},
+                                                                    'tagg': {'partitions': 5, 'replication-factor': 1}
+                                                                })
+
+        self.driver = StreamsSmokeTestDriverService(test_context, self.kafka)
+        self.processor_1 = StreamsSmokeTestJobRunnerService(test_context, self.kafka)
+        self.processor_2 = StreamsSmokeTestJobRunnerService(test_context, self.kafka)
+        self.processor_3 = StreamsSmokeTestJobRunnerService(test_context, self.kafka)
```

Comment:
Don't we need to set version 0.10.1 for those initially? Otherwise, they will have trunk version

Reply:
on line 84 of `upgrade_and_verify_start` method, we set the version with the first version from the array and start the node, so they all start with earliest specified.